src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -248,27 +224,23 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
             }
 
             if (error) {
-                reject(error);
+                return reject(error);
             }
 
-            if (paymentMethod.method === StripeElementType.CreditCard) {
-                resolve(stripePaymentMethod);
+            if (stripePaymentMethod && paymentMethod.method === StripeElementType.CreditCard) {
+                return resolve(stripePaymentMethod);
```

Comment:
I just realised this method is `async`, so you don't really need to wrap it inside a Promise and use `resolve/reject`. Async methods automatically wrap the returned value in a promise. If you want to reject a value, you should throw an exception.